### PR TITLE
Update pkg descriptions for azure-servicebus to include pointer to new package

### PR DESCRIPTION
**Pull request by ramya-rao-a**
This PR updates the package description for azure-servicebus package to include pointer to the new package and the migration guide

As discussed offline, this is in an attempt to guide users to the new package where there is active development while the above packages continue to get critical bug fixes

### CODE REVIEW

File: eng/versioning/version_data.txt [hunk 1]
```diff
@@ -34,7 +34,7 @@ com.microsoft.azure:azure-keyvault-cryptography;1.2.4;1.3.0-beta.1
 com.microsoft.azure:azure-keyvault-extensions;1.2.4;1.3.0-beta.1
 com.microsoft.azure:azure-keyvault-test;1.2.3;1.2.4
 com.microsoft.azure:azure-keyvault-webkey;1.2.4;1.3.0-beta.1
-com.microsoft.azure:azure-servicebus;3.6.2;3.7.0-beta.1
+com.microsoft.azure:azure-servicebus;3.6.2;3.6.3
```
Comment (alzimmermsft) [inline eng/versioning/version_data.txt:37]: Is the plan to ship this once this is merged?

**Review by alzimmermsft — APPROVED** [review]

File: eng/versioning/version_data.txt (hunk 1 above)
Comment (ramya-rao-a) [reply eng/versioning/version_data.txt:37]: Yes.

**Review by yvgopal — APPROVED** [review]

**Comment by ramya-rao-a** [comment]
@yvgopal, @alzimmermsft Apologies, I missed updating the description in the pom.xml file. Can you please take another look? Commit: https://github.com/Azure/azure-sdk-for-java/pull/20678/commits/5880c8797e59ecef0a476b8cff60b8c4e5cdbf4c

File: sdk/servicebus/microsoft-azure-servicebus/pom.xml [hunk 2]
```diff
@@ -6,10 +6,10 @@
   <modelVersion>4.0.0</modelVersion>
   <groupId>com.microsoft.azure</groupId>
   <artifactId>azure-servicebus</artifactId>
-  <version>3.7.0-beta.1</version> <!-- {x-version-update;com.microsoft.azure:azure-servicebus;current} -->
+  <version>3.6.3</version> <!-- {x-version-update;com.microsoft.azure:azure-servicebus;current} -->
 
   <name>Microsoft Azure SDK for Service Bus</name>
-  <description>Java library for Azure Service Bus</description>
+  <description>Please note, a newer package com.azure:azure-messaging-servicebus for Azure Service Bus is available as of December 2020. While this package will continue to receive critical bug fixes, we strongly encourage you to upgrade. Read the migration guide at https://aka.ms/azsdk/java/migrate/sb for more details.</description>
```
Comment (yvgopal) [inline sdk/servicebus/microsoft-azure-servicebus/pom.xml:12]: Keep "Java library for Azure Service Bus", also add the note.

Comment (ramya-rao-a) [reply sdk/servicebus/microsoft-azure-servicebus/pom.xml:12]: Fair point

```suggestion
  <description>Java library for Azure Service Bus. Please note, a newer package com.azure:azure-messaging-servicebus for Azure Service Bus is available as of December 2020. While this package will continue to receive critical bug fixes, we strongly encourage you to upgrade. Read the migration guide at https://aka.ms/azsdk/java/migrate/sb for more details.</description>
```

**Pull request merged**